### PR TITLE
internal: Sync from downstream

### PR DESCRIPTION
**Pull request by lnicola**

### CODE REVIEW

**Comment by lnicola** [comment]
Not sure what's happening here. The two tests (`resolve_proc_macro` and `out_dirs_check`) also fail for me on `master`, when running `CI=1 cargo +nightly test --features sysroot-abi`. CI passes because we don't run them most of the time.

**Comment by lnicola** [comment]
Huh, it's now working after upgrading to the latest nightly.

**Comment by Veykril** [comment]
It looks like `internal_features` was set to deny by default now, so we need to allow that on the test fixture's builds script part of the failing test?

**Comment by Veykril** [comment]
This is the one part thats annoying about upstream CI not running r-a tests, because this is the one integration test we do want to run ...

**Comment by lnicola** [comment]
I can reproduce the hang on `nightly`. I think we didn't spot this because we usually run `analysis-stats` on `stable` on CI.

CC @HKalbasi

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -86,6 +86,12 @@ jobs:
       - name: Test
         run: cargo test ${{ env.USE_SYSROOT_ABI }} -- --nocapture --quiet
 
+      - name: Switch to stable toolchain
```
Comment (lnicola) [inline .github/workflows/ci.yaml:89]: @Veykril heads-up, just so you're aware of this change.

**Pull request merged**